docs: Improve CONTRIBUTING.md (#19732)
ankidroid/Anki-Android: CONTRIBUTING.md

| Has no associated PR                     | `-linked:pr`                | Two people working on an issue leads to wasted effort<br/>If a linked PR is closed, it is likely that the issue will be open for contributions.                                                   |
| Has not been updated in the last 30 days | `updated:<@today-30d`       | If the issue is assigned/has a linked PR, ping for a status update if you'd want to work on the issue                                                                                             |

[#13283: Add tests to the code](<https://github.com/ankidroid/Anki-Android/issues/13283>) is also an
 extremely useful place to start.

Once you have found an issue, please make a comment asking for it to be assigned to you. 
If the issue is unassigned, assume it is workable and begin work.

## Making progress

All issues are different. Use your judgment to decide if these steps are required. 

> 1. Reproduce the issue. Comment with the version number/debug info if you're questing whether it's 

Reviewer: Is this questing or questioning?

Author: Rewritten

| Is a 'Good First Issue'                  | `label:"Good First Issue!"` | These issues should be well-documented and workable, but are often assigned and completed quickly.                                                                                                |
| Is not assigned to a user                | `no:assignee`               | Two people working on an issue leads to wasted effort<br/>Issues are commonly assigned, but not worked on. If it's been a couple of weeks, feel free to comment asking if it's still in progress. |
| Has no associated PR                     | `-linked:pr`                | Two people working on an issue leads to wasted effort<br/>If a linked PR is closed, it is likely that the issue will be open for contributions.                                                   |
| Has not been updated in the last 30 days | `updated:<@today-30d`       | If the issue is assigned/has a linked PR, ping for a status update if you'd want to work on the issue                                                                                             |

[#13283: Add tests to the code](<https://github.com/ankidroid/Anki-Android/issues/13283>) is also an

>  extremely useful place to start.

Author: I'll leave it with the justification that we want at least one 'long-running' ticket which people can continue to chip away at with multiple Pull Requests. 

It's useful to consistently be there, and it's an issue which providing value

Agreed that it's not the best, but it slightly reduces the burden on us to produce high quality 'Good First Issues'

3. Propose solution(s)
   * If you believe there are multiple reasonable solutions to the issue, post a comment documenting
    them, feel free to link the proposal on Discord

When posting comments, aim to be clear and brief. Use `<details>` if posting large images, or blocks
 of text.

## Before writing code (one-time)

> [!IMPORTANT]
> `git clone` your fork and define `origin` and `upstream`: [Wiki - Repository Setup](https://github.com/ankidroid/Anki-Android/wiki/Development-Guide/#initial-setup-one-time) 

> [!TIP]
> Ensure that `user.email` is associated with your GitHub account, so your commits appear on your heatmap  
> https://docs.github.com/en/account-and-profile/how-tos/email-preferences/setting-your-commit-email-address

> [!TIP] 
> Keep `main` on your fork in sync with `upstream/main` when starting new work
> If you make a commit to `main`, make a new branch to retain the commit, then `git rebase` and drop
> your commit from `main`.

Ensure you branched off `main` recently and are submitting from a branch on your fork (not `main`). 

```sh
git checkout main
git pull upstream main
git checkout -b my-feature-branch

> # make your changes to the source code

Reviewer: just a suggestion but maybe could add to this comment with something like "make your changes to the source code then add and commit", if someone hasn't used git before they wouldn't know to add and commit and would probably be confused. For the first few times when I was following those instructions, I always forgot to add and commit since I wasn't used to git

## Before submitting a Pull Request (PR)

> [!WARNING]
> Submitting a PR from `main` makes it difficult to submit multiple pull requests and makes it
> difficult for your fork of `main` to stay in sync with `upstream`

We have a limited amount of time to spend on the project. We are trying our best.
 A PR which is well-formed saves us time, and will be merged more quickly.
 Unless requested, PRs should be sent in a form where you feel they should immediately be merged.
  
1. Please run these checks before submitting a PR: [workflows/README.md](https://github.com/ankidroid/Anki-Android/tree/main/.github/workflows/README.md#quality-checks). Your PR will not be reviewed until automated checks 
   using GitHub Actions pass.
2. Check commits/diffs
    * Each commit should compile, and have one purpose
        * Rebase and add additional commits if a change contains multiple purposes
    * No merge commits in the history
    * The 'Files Changed' tab of the PR should only contain relevant changes
        * Incidental changes (lint, refactorings) should be in separate commits
        * Whitespace changes should be reverted  
3. Write a PR description

>     * Our [PR template]((https://github.com/david-allison/Anki-Android/blob/contributing-md/.github/pull_request_template.md)) **must** be filled (unless trivial). Create PRs using GitHub so this template will be pre-populated for you to submit.

Reviewer 2: ```suggestion
    * Our [PR template]((https://github.com/david-allison/Anki-Android/blob/contributing-md/.github/pull_request_template.md)) **must** be filled (unless trivial). Create PRs using GitHub so this template will be present in the PR Description area, ready for you to fill in with information about your work.
```

> Writing automated tests helps us with confident that your change is correct. This is currently 

Reviewer 2: ```suggestion
Writing automated tests helps reviewers be more confident that your change is correct. This is currently 
```

 optional but recommended, and should reduce the number of rounds of reviews.
 See also: [Wiki - Testing guide](https://github.com/ankidroid/Anki-Android/wiki/Testing).

## Handling requests for changes

> [!IMPORTANT]
> Don't close a PR if something unexpected happened, use `git push --force` to fix it, or get in touch.

AnkiDroid expects use of `git rebase` and `git push --force` to PR branches. PRs with merge commits 
 will not be merged.

Any mistakes made in commits should be rebased out of the history before merge.

See: https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

### Comments/Suggestions/Patches

Most changes are requested via **comments** on a Pull Request.
 You do not need to write a reply to all comments, a +1 reaction that the comment is handled 
 is sufficient.
If a comment is confusing, please ask for clarification: you should aim to understand the 'why' of a
 review comment, rather than following the comment without thinking about the impact.

If a **suggestion** is provided, feel free to apply it in the GitHub UI. Ideally `rebase` & `squash` it 

>  locally and `force push` the branch to keep the history clean

Reviewer 2: ```suggestion
If a **suggestion** is provided, feel free to apply it in the GitHub UI. Then ideally you pull those new commits down via`git rebase` or `git pull` and then you `squash` the changes into the relevant commits 
 locally and `force push` the branch to keep the history clean
```

  * Awaiting review from a reviewer
* **[Needs Second Approval](https://github.com/ankidroid/Anki-Android/labels/Needs%20Second%20Approval)**
  * Awaiting a review from another reviewer
* **[Pending Merge](https://github.com/ankidroid/Anki-Android/labels/Pending%20Merge)**
  * No action is needed. Awaiting potential additional action, then a marge by a maintainer.
* **[Blocked by dependency](https://github.com/ankidroid/Anki-Android/labels/Blocked%20by%20dependency)**
  * The PR will not be merged until another issue/PR is merged. Please ping if you don't understand 
    why this occurs. 
* **[Strings](https://github.com/ankidroid/Anki-Android/labels/Strings)**
  * No action is needed. Maintainers need to perform extra steps once the PR is 'Pending Merge'. 
* **[squash-merge](https://github.com/ankidroid/Anki-Android/labels/squash-merge)**
  * If you squash merge your PR, it will be mergeable, otherwise a maintainer will squash merge it 
    when it moves to the 'Pending Merge' state

## After a PR is waiting for reviewer feedback

Feel free to select a new issue. Once you've done a handful of '**Good First Issues**', we hope you 

>  feel that most of the issues in the repo are within your grasp. 

Reviewer 3: > Feel free to select a new issue _but do focus on resolving review requests on your previous PRs to get those merged as fast as possible_. Once you've done a handful of '**Good First Issues**', we hope you feel that most of the issues in the repo are within your grasp.

Good First Issues take time to write, we don't want experienced contributors taking on too many, 
 as there are likely higher impact issues to resolve, and it makes it harder for new contributors 

>  to find something to get started with

Reviewer 3: > Good First Issues take time to write, we don't want experienced contributors taking on too many, 
 as there are likely higher impact issues to resolve, and it makes it harder for new contributors 
 to find something to get started with

=>

> Good
